modules/pubmaticBidAdapter.js

```diff
@@ -217,6 +219,108 @@ function _checkParamDataType(key, value, datatype) {
   }
 }
 
+function _createNativeRequest(params) {
+  var nativeRequestObject = {
+    assets: []
+  };
+  for (var key in params) {
+    var assetObj = {};
+    if (!(nativeRequestObject.assets && nativeRequestObject.assets.length > 0 && nativeRequestObject.assets.hasOwnProperty('key'))) {
+      switch (key) {
+        case 'title':
+          assetObj = {
+            id: 1,
+            req: params[key].required,
+            title: {
+              len: params[key].length,
+              ext: params[key].ext
+            }
+          };
+          break;
+        case 'image':
+          assetObj = {
+            id: 2,
+            req: params[key].required,
+            img: {
+              type: params[key].type,
+              w: params[key].w || params[key].width || params[key].sizes[0],
+              h: params[key].h || params[key].height || params[key].sizes[1],
+              wmin: params[key].wmin || params[key].minimumWidth || (params[key].minsizes ? params[key].minsizes[0] : undefined),
+              hmin: params[key].hmin || params[key].minimumHeight || (params[key].minsizes ? params[key].minsizes[1] : undefined),
+              mimes: params[key].mimes,
+              ext: params[key].ext
+            }
+          };
+          break;
+        case 'icon':
+          assetObj = {
+            id: 3,
+            req: params[key].required,
+            icon: {
+              type: params[key].type,
+              w: params[key].w || params[key].width || (params[key].sizes ? params[key].sizes[0] : undefined),
+              h: params[key].h || params[key].height || (params[key].sizes ? params[key].sizes[1] : undefined)
+            }
+          };
+          break;
+        case 'sponsoredBy':
+          assetObj = {
+            id: 4,
+            req: params[key].required,
+            sponsoredBy: {}
+          };
+          break;
+        case 'body':
+          assetObj = {
+            id: 5,
+            req: params[key].required,
+            desc: {}
+          };
+          break;
+        case 'clickUrl':
+          assetObj = {
+            id: 6,
+            req: params[key].required,
+            clickUrl: {}
```

Comment:
is it necessary to send empty object in the request? like clickUrl: {}

Reply:
No, 
i was sending it incorrectly 
Ad server ignores it 
Removed it as clickUrl is not accepted by adServer